### PR TITLE
Added Environments

### LINKED ISSUE #166: Add Environments
Go live in public preview on Dec 15th
https://devblogs.microsoft.com/devops/i-need-manual-approvers-for-github-actions-and-i-got-them-now/

### PR DESCRIPTION
**Pull request by samsmithnz**
- Addresses #166 

Awaiting mid-Dec release of Actions with this feature to confirm functionality

### CODE REVIEW

File: src/AzurePipelinesToGitHubActionsConverter.Core/PipelinesToActionsConversion/GeneralProcessing.cs [hunk 1]
```diff
@@ -85,17 +85,17 @@ public AzurePipelines.Environment ProcessEnvironmentV2(string environmentYaml)
                             string resourceName = null;
                             if (json["resourceName"] != null)
                             {
-                                name = json["resourceName"].ToString();
```
Comment (samsmithnz) [inline src/AzurePipelinesToGitHubActionsConverter.Core/PipelinesToActionsConversion/GeneralProcessing.cs:88]: Oops - found a bug. Fixed now :)

File: src/AzurePipelinesToGitHubActionsConverter.Core/AzurePipelinesToGitHubActionsConverter.Core.csproj [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (f2calv) [inline src/AzurePipelinesToGitHubActionsConverter.Core/AzurePipelinesToGitHubActionsConverter.Core.csproj:4]: surely its time to multi-target for .net 5.0 ? :)

File: src/AzurePipelinesToGitHubActionsConverter.Core/AzurePipelinesToGitHubActionsConverter.Core.csproj [hunk 3]
```diff
@@ -2,7 +2,7 @@
...
-    <Version>0.13.1.0</Version>
```
Comment (f2calv) [inline src/AzurePipelinesToGitHubActionsConverter.Core/AzurePipelinesToGitHubActionsConverter.Core.csproj:5]: someone I know has a great nuget publish pipeline...

File: src/AzurePipelinesToGitHubActionsConverter.Core/PipelinesToActionsConversion/GeneralProcessing.cs [hunk 4]
```diff
@@ -86,17 +86,17 @@ public AzurePipelines.Environment ProcessEnvironmentV2(string environmentYaml)
...
+                                resourceName = json["resourceName"].ToString();
```
Comment (f2calv) [inline src/AzurePipelinesToGitHubActionsConverter.Core/PipelinesToActionsConversion/GeneralProcessing.cs:89]: avoid 'magic strings', 'resourceName = json[nameof(resourceName).ToString();'

Comment (samsmithnz) [reply src/AzurePipelinesToGitHubActionsConverter.Core/PipelinesToActionsConversion/GeneralProcessing.cs:89]: I don't know what 'magic strings' means. Please explain.

File: src/AzurePipelinesToGitHubActionsConverter.Core/AzurePipelinesToGitHubActionsConverter.Core.csproj (hunk 3 above)
Comment (samsmithnz) [reply src/AzurePipelinesToGitHubActionsConverter.Core/AzurePipelinesToGitHubActionsConverter.Core.csproj:5]: #126

File: src/AzurePipelinesToGitHubActionsConverter.Core/PipelinesToActionsConversion/GeneralProcessing.cs (hunk 4 above)
Comment (f2calv) [reply src/AzurePipelinesToGitHubActionsConverter.Core/PipelinesToActionsConversion/GeneralProcessing.cs:89]: Anything inside quotes is outside the scope of the compiler, therefore when parameters/arguments/etc get renamed then you get instances where fieldOldName = "fieldNewName" happen which can make code harder to follow/debug.
Basically removing magic strings leads to tighter coupling between code and API.
https://softwareengineering.stackexchange.com/questions/365339/what-is-wrong-with-magic-strings

Comment (samsmithnz) [reply src/AzurePipelinesToGitHubActionsConverter.Core/PipelinesToActionsConversion/GeneralProcessing.cs:89]: Ahh got it. So use CONST instead. I'll think about it. Migration apps are a bit different than a regular app and resource strings/etc. It's a lot of work in this app (I have a lot of magic strings now that I'm parsing the JSON), for something that doesn't add much value, in my opinion.  I do appreciate the idea though. I'll keep it in mind.

**Pull request merged**